core/src/platform.h

```diff
@@ -33,3 +38,6 @@ std::string stringFromResource(const char* _path);
  * allocated file
  */ 
 unsigned char* bytesFromResource(const char* _path, unsigned int* _size);
+
+bool fetchData(std::unique_ptr<std::string> _url, std::stringstream& _rawData);
```

Comment:
Using a `unique_ptr` for the url seems unnecessary, in every implementation you simply grab the `c_str()` so you may as well pass one in and make it less verbose